### ISSUE TITLE
How can I do the decoding only base on the acoustic models where there is no effect of language models in EESEN ?

### ISSUE THREAD

**Issue by niucheney** [srvk/eesen#118]
Hi,
I just want to know which frame the CTC tookit (EESEN) chooses when decoding. Can EESEN  output it?
I am looking forward to your reply.
Best wishes!

**Comment by fmetze** [comment]
You can simply look at the output of net-output-extract, which is a matrix of number of frames by number of tokens. You can see it being passed to the decoder (which applies the language model) in decode_ctc_lat.sh (or other decoding scripts) as a pipe, where you can store it or otherwise process it (convert to text from using copy-feats or so).

**Comment by niucheney** [comment]
Thanks for your reply. 
Perhaps I don't explain it clearly. I have output the posterior probabilities of tokens using net-output-extract and copy-feats. When ploting the posterior probabilities of tokens, there is a series of spikes or peaks separated by blanks according to the paper (Alex Graves et al,Connectionist Temporal Classification: Labelling Unsegmented Sequence Data with Recurrent Neural Networks, 2006).  I just want to know how to locate the positions of spikes in EESEN.  Is there a way to get it from lattice?

**Comment by fmetze** [comment]
Look at ErrorRateMSeq in https://github.com/srvk/eesen/blob/blank_scale_and_parallel_models/src/net/ctc-loss.cc, you can call train-ctc-parallel in validation mode with the "--sequence-out-file" option, which will give you a file that contains the IDs, temporal locations, and confidences of the peaks. This is what you want, right?

We'll integrate this into the main branch soon (hopefully).

**Comment by niucheney** [comment]
@fmetze Thank you very much for the prompt reply. The function of the option is very cool !
